src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -324,15 +324,57 @@ private bool MatchRef(int index, int length, ReadOnlySpan<char> inputSpan)
 
         private void Backwardnext() => runtextpos += _rightToLeft ? 1 : -1;
 
-        protected override bool FindFirstChar() =>
-            _code.FindOptimizations.TryFindNextStartingPosition(runtext!, ref runtextpos, runtextbeg, runtextstart, runtextend);
+        protected internal override void Scan(ReadOnlySpan<char> text)
+        {
+            // Configure the additional value to "bump" the position along each time we loop around
+            // to call FindFirstChar again, as well as the stopping position for the loop.  We generally
+            // bump by 1 and stop at textend, but if we're examining right-to-left, we instead bump
+            // by -1 and stop at textbeg.
+            int bump = 1, stoppos = text.Length;
+            Debug.Assert(runregex != null);
+            if (runregex.RightToLeft)
+            {
+                bump = -1;
+                stoppos = 0;
+            }
+
+            while (true)
+            {
+                if (FindFirstChar(text))
+                {
+                    CheckTimeout();
+
+                    Go(text);
+
+                    // If we got a match, we're done.
+                    if (runmatch!._matchcount[0] > 0)
+                    {
+                        return;
+                    }
```

Comment:
Same here.  Can Go return a bool?

Reply:
Yeah, it can. Here it wouldn't really be a perf gain like in the compiled engine, just consistency between engines. Interpreted doesn't need it as it can access `_matchcount` field, but I'm fine with moving all of them for consistency sake.